components/ai_chat/resources/page/components/smart_mode_modal/smart_mode_modal.tsx

```diff
@@ -0,0 +1,347 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Dialog from '@brave/leo/react/dialog'
+import Dropdown from '@brave/leo/react/dropdown'
+import Icon from '@brave/leo/react/icon'
+import Input from '@brave/leo/react/input'
+import Label from '@brave/leo/react/label'
+import TextArea from '@brave/leo/react/textarea'
+import classnames from '$web-common/classnames'
+import { getLocale } from '$web-common/locale'
+import styles from './smart_mode_modal_style.module.scss'
+import { useAIChat } from '../../state/ai_chat_context'
+import { useConversation } from '../../state/conversation_context'
+import { getModelIcon } from '../../../common/constants'
+import * as Mojom from '../../../common/mojom'
+
+export default function SmartModeModal() {
+  const aiChatContext = useAIChat()
+  const conversationContext = useConversation()
+
+  // State
+  const [selectedModel, setSelectedModel] = React.useState(
+    aiChatContext.smartModeDialog?.model,
```

Comment:
Hey just curious - why do these properties need to live in `AIChatContext`? Seems like they'd only be used here?

Reply:
(not saying this is wrong)

Reply:
Needs to be in `context` since, we have to set these values from inside the `untrusted-iframe` as well to be set from the `tools menu`